src/main/java/io/r2dbc/pool/ConnectionPool.java

```diff
@@ -209,15 +209,13 @@ private InstrumentedPool<Connection> createConnectionPool(ConnectionPoolConfigur
         // current implementation. (https://github.com/reactor/reactor-pool/issues/33)
         // To workaround the issue, here defines an evictionPredicate that performs both maxIdleTime and maxLifeTime check.
         BiPredicate<Connection, PooledRefMetadata> evictionPredicate = (connection, metadata) -> {
-            long maxIdleTimeMills = maxIdleTime.toMillis();
-            long maxLifeTimeMillis = maxLifeTime.toMillis();
-
-            if (maxIdleTimeMills == 0 || maxLifeTimeMillis == 0) {
+            if (maxIdleTime.isZero() || maxLifeTime.isZero()) {
+                // evict immediately
```

Comment:
> just a small suggestion, as I've stumbled upon this bug myself, I think it's easier to just replace if (maxIdleTimeMills == 0 || maxLifeTimeMillis == 0) with if (maxIdleTime.isZero() || maxLifeTime.isZero()) in the evictionPredicate, this way it's just one line of code changed, the code is easier to read and all tests will pas

@GabrielCalin isn't that exactly what I changed here?

Reply:
You're right, I'm not sure why I didn't see it. Hopefully someone will merge it soon. I had to add a very big maximum lifetime at the moment as a workaround.

Reply:
yes, I'm using that workaround as well. Also note that you have to set a `backgroundEvictionInterval`, otherwise there will be no eviction at all

Reply:
Oh, thanks, so `maxIdleTime` & 'maxLifeTime' even if explicitly set are not working for background eviction with the default `backgroundEvictionInterval`? (I assume because of the same issue)

Reply:
> You're right, I'm not sure why I didn't see it. Hopefully someone will merge it soon. I had to add a very big maximum lifetime at the moment as a workaround.

Isn't `-1` the maximum lifetime value now?

Reply:
> > ... I had to add a very big maximum lifetime at the moment as a workaround.
> 
> Isn't `-1` the maximum lifetime value now?

Because of the bug, `-1 `gets assimilated to `0`, which means immediate eviction. It's actually why the default value of `-1` doesn't work as expected and instead causes connections to be dropped before each query. The workaround is to set it to a big value, in practice meaning the same as `-1` is supposed to mean, that connections are not expired.

Reply:
Hmm, setting maxLifeTime to -1 actually helped us. With the default maxLifeTime, all connections were instantly expiring, while with -1 they stay forever.